chat/src/main/java/io/skygear/plugins/chat/MessageCacheObject.java

```diff
@@ -72,13 +72,35 @@ public MessageCacheObject(Message message) {
         this.conversationID = message.getConversationId();
         this.creationDate = message.getCreatedTime();
         this.editionDate = (Date) message.record.get("edited_at");
-        this.deleted = (Boolean) message.record.get("deleted");
+        Boolean deleted = (Boolean) message.record.get("deleted");
+        if (deleted == null) {
+            this.deleted = false;
+        } else {
+            this.deleted = deleted;
+        }
         this.jsonData = message.toJson().toString();
+        this.sendDate = message.sendDate;
+        this.alreadySyncToServer = message.alreadySyncToServer;
+        this.fail = message.fail;
+
+        // creationDate of the record originally represents the message creation date on server
```

Comment:
nice comment

Reply:
oh, this reminds me to update the terminology since i copied it from the ios directly

Reply:
...